deploy-service/common/src/main/java/com/pinterest/deployservice/scm/GithubManager.java

```diff
@@ -85,15 +85,19 @@ private void setHeaders() throws Exception {
                 GitHub gitHubApp = new GitHubBuilder().withJwtToken(jwtToken).build();
                 GHAppInstallation appInstallation = gitHubApp.getApp().getInstallationByOrganization(this.githubAppOrganization);
                 GHAppInstallationToken appInstallationToken = appInstallation.createToken().create();    
-                this.token = appInstallationToken.getToken();
+
+                // always use the newly created github app token as the token will expire
+                this.headers.put("Authorization", String.format("Token %s", appInstallationToken.getToken()));
             } catch (Exception e) {
                 // e.printStackTrace();
                 LOG.error("Exception when getting Github token: ", e);
                 throw e;
             }
         }
-
-        this.headers.put("Authorization", String.format("Token %s", this.token));
```

Comment:
what's the issue with original code? 

Before:
--------
if (token empty):
  get token

set header

After:
------
if (token empty):
  get token
  set header
else
  set header